src/Shared/NodeEndpointOutOfProcBase.cs

```diff
@@ -115,6 +116,8 @@ internal abstract class NodeEndpointOutOfProcBase : INodeEndpoint
         /// </summary>
         private BinaryWriter _binaryWriter;
 
+        private readonly IList<string> _versionHandshakeGroup = ["fileVersionMajor", "fileVersionMinor", "fileVersionBuild", "fileVersionPrivate"];
```

Comment:
```suggestion
        private readonly ImmutableArray<string> _versionHandshakeGroup = ["fileVersionMajor", "fileVersionMinor", "fileVersionBuild", "fileVersionPrivate"];
```

Given this collection doesn't change consider a non-mutating type to make future mutations an explicit decision.